keyvault/client/secrets/src/main/java/com/azure/keyvault/SecretAsyncClient.java

```diff
@@ -0,0 +1,473 @@
+     * This operation requires the {@code secrets/set} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Creates a new secret in the key vault. Subscribes to the call asynchronously and prints out the newly
+     * created secret details when the callback happens.</p>
+     * <pre>
+     * secretAsyncClient.setSecret("secretName", "secretValue").subscribe(secretResponse -&gt;
+     *   System.out.printf("Secret is created with name %s and value %s \n", secretResponse.value().name(), secretResponse.value().value()));
+     * </pre>
+     *
+     * @param name The name of the secret. It is required and cannot be null.
+     * @param value The value of the secret. It is required and cannot be null.
+     * @throws ResourceModifiedException if invalid {@code name} or {@code value} are specified.
+     * @throws HttpRequestException if {@code name} or {@code value} are empty strings.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Secret created secret}.
+     */
+    public Mono<Response<Secret>> setSecret(String name, String value) {
+        SecretRequestParameters parameters = new SecretRequestParameters().value(value);
+        return service.setSecret(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * Get the specified secret with specified version from the key vault. The get operation is applicable to any secret stored in Azure Key Vault.
+     * This operation requires the {@code secrets/get} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Gets a specific version of the secret in the key vault. Subscribes to the call asynchronously and prints out the
+     * returned secret details when the callback happens.</p>
+     * <pre>
+     * String secretVersion = "6A385B124DEF4096AF1361A85B16C204";
+     * secretAsyncClient.getSecret("secretName", secretVersion).subscribe(secretResponse -&gt;
+     *   System.out.printf("Secret with name %s, value %s and version %s", secretResponse.value().name(),
+     *   secretResponse.value().value(), secretResponse.value().version()));
+     * </pre>
+     *
+     * @param name The name of the secret, cannot be null
+     * @param version The version of the secret to retrieve. If this is an empty String or null, this call is equivalent to calling {@link #getSecret(String)}, with the latest version being retrieved.
+     * @throws ResourceNotFoundException when a secret with {@code name} and {@code version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@code name}  name} or {@code version} are empty strings.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the requested {@link Secret secret}.
+     */
+    public Mono<Response<Secret>> getSecret(String name, String version) {
+        String secretVersion = "";
+        if(version != null){
```

Comment:
Missing spaces here - checkstyle should be run?

Reply:
fixed